test/api/test_api.cpp

```diff
@@ -139,6 +139,16 @@ static void parallel_query(Connection *conn, bool *correct, size_t threadnr) {
 	}
 }
 
+TEST_CASE("Test temp_directory defaults", "[api][.]") {
+	const char *db_paths[] = {nullptr, "", ":memory:", ":memory:named_conn"};
+	for (auto &path : db_paths) {
+		auto db = make_uniq<DuckDB>(path);
+		auto conn = make_uniq<Connection>(*db);
+
+		REQUIRE(db->instance->config.options.temporary_directory == ".tmp");
```

Comment:
Maybe the default can be something like `duckdb_memory.[pid].tmp`?

Reply:
The comment after this supersedes this right?
We ideally just don't set a temp directory for an in-memory database, that seems unexpected